### PR TITLE
[FLINK-16921][e2e] Make Kubernetes e2e test stable

### PR DESCRIPTION
**Pull request by wangyang0918**
## What is the purpose of the change

Use the following optimization to make this K8s e2e test more stable both in azure/travis and local.

* Remove start/stop code for non-linux environment, let the user manually start/stop minikube. Since we will encounter various problem in local environment(e.g. Mac), including disk problem, network problem, docker daemon, driver setup(virtualbox, etc.). 
* Print more debug information so that when the test failed it is easier to investigate
* Explicitly wait for the dispatcher running and then submit the job to the existing cluster

## Brief change log
* Do not start/stop minikube in non-linux environment for k8s e2e tests
* Print more information for debugging when Kubernetes e2e tests failed
* Wait for rest endpoint up and then submit Flink job to existing Kubernetes session

## Verifying this change

* Run in local environment(Mac) more than 10 times, all should pass
* Run in azure more that 3 times, all should pass

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

File: flink-end-to-end-tests/test-scripts/common_kubernetes.sh [hunk 1]
```diff
@@ -88,22 +86,43 @@ function start_kubernetes_if_not_running {
...
+        if ! retry_times ${MINIKUBE_START_RETRIES} ${MINIKUBE_START_BACKOFF} "${stop_command}"; then
+            echo "Could not stop minikube. Aborting..."
+            exit 1
+        fi
     fi
 }
 
+function debug_copy_and_show_logs {
+    echo "Debugging failed Kubernetes test:"
+    echo "Currently exiting Kubernetes resources"
```
Comment (rmetzger) [inline flink-end-to-end-tests/test-scripts/common_kubernetes.sh:119]:
```suggestion
    echo "Currently existing Kubernetes resources"
```

Comment (wangyang0918) [reply flink-end-to-end-tests/test-scripts/common_kubernetes.sh:119]: Nice catch. I will fix it.

**Comment by wangyang0918** [comment]
@rmetzger It seems that the azure pipeline is cancelled by timeout(4h). The two Kubernetes related tests could always pass in the recent two azure run.

I also have run the test locally more that 10 times. It could always pass.

**Comment by rmetzger** [comment]
The e2e failure is known: https://issues.apache.org/jira/browse/FLINK-16423

Since the failing e2e test is running after the kubernetes test, I'm +1 to merge this PR.

**Comment by rmetzger** [comment]
I'm merging the PR now.

**Pull request merged**